projects/packages/search/src/dashboard/components/record-meter/index.jsx

```diff
@@ -15,11 +22,18 @@ import './style.scss';
  * @returns {React.Component} RecordMeter React component
  */
 export default function RecordMeter() {
+	const tierMaximumRecords = useSelect( select => select( STORE_ID ).getTierMaximumRecords() );
```

Comment:
I understand that by using state store we don't have to pass down props for nested components which makes state management easier. However, by integrating with the state store, the component becomes less reusable. I'm not sure about to what degree shall we balance that? 😄

Reply:
@kangzj would your instinct be to have the main dashboard component do all the state stuff and pass that down in props? I couldn't decide either 🤔